hack/test/e2e-run.sh

```diff
@@ -1,41 +1,72 @@
 #!/usr/bin/env bash
-set -e
+set -e -o pipefail
 
-TESTFLAGS=${TESTFLAGS:-""}
-# Currently only DockerSuite and DockerNetworkSuite have been adapted for E2E testing
-TESTFLAGS_LEGACY=${TESTFLAGS_LEGACY:-""}
-TIMEOUT=${TIMEOUT:-60m}
+ARCH=$(uname -m)
+if [ "$ARCH" == "x86_64" ]; then
+  ARCH="amd64"
+fi
 
-SCRIPTDIR="$(dirname ${BASH_SOURCE[0]})"
+export DOCKER_ENGINE_GOARCH=${DOCKER_ENGINE_GOARCH:-${ARCH}}
 
-export DOCKER_ENGINE_GOARCH=${DOCKER_ENGINE_GOARCH:-amd64}
+# Set defaults
+: ${TESTFLAGS:=}
+: ${TESTDEBUG:=}
+
+integration_api_dirs=${TEST_INTEGRATION_DIR:-"$(
+	find ./integration -type d |
+	grep -vE '(^./integration($|/internal)|/testdata)')"}
 
 run_test_integration() {
-  run_test_integration_suites
-  run_test_integration_legacy_suites
+	[[ "$TESTFLAGS" != *-check.f* ]] && run_test_integration_suites
+	run_test_integration_legacy_suites
```

Comment:
hopefully we can find a way to share this without having to duplicate it once we do more work on #35937